### DISCUSSION TITLE
Laragon automatically entering to /public directory.

### DISCUSSION THREAD

**Discussion by barancihan** [leokhoa/laragon#430] [Q&A]
Hello guys im beginner on laragon and my projects are codeigniter. Laragon is automatically adding /public to DocumentRoot. I removed it and laragon still adding it. Is there anyone know how to fix it ?

**Comment by Pen-y-Fan** [comment]
Any Apache **auto.<site>.<domain>.conf** file is automatically overwritten when changes are made to Laragon. Laragon will automatically check if the project has a **/public** directory and if there is one will make this the DocumentRoot. This automatic check can not be set in properties. The workaround is to create your own manual version of the file. 

Example project:

- the project is **codeigniter**
- the domain is **test**

Create a new apache conf file for codeigniter.local:

Laragon menu > Apache > dir:conf, open the **sites-enabled** directory (this should be similar to C:\laragon\etc\apache2\sites-enabled)

Copy the existing auto generated file, for your project e.g.:

```text
auto.codeigniter.test.conf
```

Auto files starting **auto** are overwritten, therefore create a file with a different prefix. Paste and rename **auto** to **manual** and **test** to **local**, e.g:

```text
manual.codeigniter.local.conf
```

Edit the **manual** file and amend the ROOT to remove **/public** and SITE to codeigniter.local

```text
define ROOT "C:/laragon/www/codeigniter"
define SITE "codeigniter.local"
```

Manually add the project to the host file, click the **h** in the top right of Laragon copy the laragon magic version of the project. Paste it above the **# Laragon magic** section, and rename the domain from **test** to **local**,

Please don't change the laragon magic section, Laragon will auto-create these.

```text
127.0.0.1	codeigniter.local       # Manual set see C:\laragon\www\codeigniter

# Laragon magic
127.0.0.1      codeigniter.test        #laragon magic! 
```

Stop and start Apache to reload the config and your project can be opened at <https://codeigniter.local>

I hope this helps.

P.S. As you have said you are new to Laragon, so you may find my blog helpful:

- [How to set up Laragon on a new Windows computer (part 1) - Install Laragon](https://pen-y-fan.github.io/2022/11/07/how-to-set-up-laragon-on-a-new-windows-computer/)

**Comment by FlavioSuarez** [comment]
Hi,

I did a fresh install of Laragon 6 and ran it as Administrator...
After a while, the Document root setting will be C:\laragon6\www

Then I edited the quick app config and added these 2 lines:

`# CodeIgniter

CodeIgniter=composer create-project codeigniter4/appstarter %s
`

Here it works ok...
If it don't work to you, try Menu > www > Switch Document Root

Hope this help you!
